src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -248,9 +308,14 @@ private static void mergeHeaderlessBamShards(String outputFile, SAMFileHeader he
         fs.delete(outputPath, true);
 
         try (final OutputStream out = fs.create(outputPath)) {
-            new SAMOutputPreparer().prepareForRecords(out, SAMFormat.BAM, header); // write the header
+            new SAMOutputPreparer().prepareForRecords(out, samOutputFormat, header); // write the header
```

Comment:
Is `header` here guaranteed to be the same as the one set on the corresponding `OutputFormat` class?